public/app/plugins/panel/xychart/v2/XYChartPanel.tsx

```diff
@@ -68,17 +78,82 @@ export const XYChartPanel2 = (props: Props2) => {
           getItemKey: () => `${idx}-${s.name.value}`,
           fieldName: yField.state?.displayName ?? yField.name,
           disabled: yField.state?.hideFrom?.viz ?? false,
+          getDisplayValues: () => {
```

Comment:
Is this function different than the one used in PlotLegend?

Reply:
Or in the original XYChart? Maybe we can DRY it out if it is?

Reply:
Pushed a quick update. The plan was to extract it to a separate component, but not today apparently 😅